spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2724,8 +2724,9 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           return None
         }
 
-        if ((join.leftKeys ++ join.rightKeys).exists(_.dataType.isInstanceOf[StructType])) {
-          withInfo(join, "Unsupported struct data type in join keys")
+        val keyTypes = (join.leftKeys ++ join.rightKeys).map(_.dataType)
```

Comment:
nit: could probably add a `.distinct` on the end of this line and reduce the number of calls to check supported types?

Reply:
This got removed due to this comment: https://github.com/apache/datafusion-comet/pull/735#discussion_r1714452934